docs/api-reference.md

```diff
@@ -226,6 +289,48 @@ class App {
 
 ```
 
+
+`getAuth(): FirebaseAuthState` : Returns the client's current Authentication State.
+
+Sample Usage:
+
+```ts
+constructor(public auth: FirebaseAuth) {
+        this.authState = auth.getAuth();
+        auth.subscribe((state: FirebaseAuthState) => {
+            this.authState = state;          
+        });        
+    }
+
+    get authenticated(): boolean {        
+        return this.authState !== null;
+    }
+```
+
+`logout(): void`: Deletes the authentication token issued by Firebase and signs user out. See [Auth.signOut()](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#signOut) in the Firebase API reference.
+
+Sample Usage:
+
+```ts
+	signOut(): {
+		this.af.auth.logout();
+	}
+```
+
+`createUser(credentials: EmailPasswordCredentials): firebase.Promise<FirebaseAuthState>` : Creates a new user with email/password provided. Returns a promise filled with FirebaseAuthState, which contains the uid of the created user.
+
+The credentials object is an object containing email and password of the user to be created. Similar to [createUserWithEmailAndPassword](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#createUserWithEmailAndPassword)
+
+```ts
+  createNewUser(): Promise<FirebaseAuthState> {
```

Comment:
I think this should be:  `createNewUser( properties:Object ): Promise<FirebaseAuthState> {`?